src/Vault.sol

```diff
@@ -126,6 +126,27 @@ contract Vault is IVault, VaultToken, Ownable {
         SettlementGuard.accountDelta(msg.sender, currency, -(paid.toInt128()));
     }
 
+    function settleAndRefund(Currency currency, address to)
+        external
+        payable
+        override
+        isLocked
+        returns (uint256 paid, uint256 refund)
+    {
+        paid = currency.balanceOfSelf() - reservesOfVault[currency];
+        int256 currentDelta = SettlementGuard.getCurrencyDelta(msg.sender, currency);
+
+        if (currentDelta >= 0 && paid > currentDelta.toUint256()) {
+            // msg.sender owes vault but paid more than than whats owed
+            refund = paid - currentDelta.toUint256();
+            paid = currentDelta.toUint256();
+            currency.transfer(to, refund);
```

Comment:
We need to think about whether there are re-entrant issues.
Let me think about this. maybe need to put this in the end after updated reservesOfVault and SettlementGuard.